### PR TITLE
FIX: parse msg for MsgEthereumTx

### PR DESCRIPTION
**Pull request by vincentysc**

### CODE REVIEW

File: usecase/parser/authinfo.go [hunk 1]
```diff
@@ -16,11 +16,12 @@ func ParseSignerInfosToTransactionSigners(
 	signerInfos []utils.SignerInfo,
 	accountAddressPrefix string,
 	possibleSignerAddresses []string,
+	txHash string,
 ) ([]model.TransactionSigner, error) {
 	var signers []model.TransactionSigner
 
 	if len(signerInfos) <= 0 && len(possibleSignerAddresses) <= 0 {
-		panic("error signer info not found")
+		panic(fmt.Errorf("error signer info not found at tx %q", txHash))
```
Comment (davcrypto) [inline usecase/parser/authinfo.go:24]: `Errorf` ->`Sprintf`

File: usecase/parser/msg.go [hunk 2]
```diff
@@ -1959,3 +1962,62 @@ func ParseMsgCreateVestingAccount(
...
+
+	if !parserParams.MsgCommonParams.TxSuccess {
+		msgEthereumTxParams := model.MsgEthereumTxParams{
+			RawMsgEthereumTx: rawMsg,
+		}
+
+		// Getting possible signer address from Msg
+		var possibleSignerAddresses []string
+		// TODO: get signer
+		if msgEthereumTxParams.From == "" {
```
Comment (davcrypto) [inline usecase/parser/msg.go:1996]: is it a `!=`?

File: usecase/parser/msg.go [hunk 3]
```diff
@@ -1959,3 +1962,62 @@ func ParseMsgCreateVestingAccount(
...
+	}
+
+	msgEthereumTxParams := model.MsgEthereumTxParams{
+		RawMsgEthereumTx: rawMsg,
+	}
+
+	// Getting possible signer address from Msg
+	var possibleSignerAddresses []string
+	// TODO: get signer
+	if msgEthereumTxParams.From == "" {
```
Comment (davcrypto) [inline usecase/parser/msg.go:2014]: here as well

File: usecase/parser/block.go [hunk 4]
```diff
@@ -49,7 +49,7 @@ func ParseBlockToCommands(
 			return nil, fmt.Errorf("error parsing message commands: %v", parseErr)
 		}
 
-		transactionCommands, parseErr := ParseTransactionCommands(txDecoder, cosmosClient, block, blockResults, accountAddressPrefix, possibleSignerAddresses)
+		transactionCommands, parseErr := ParseTransactionCommands(parserManager.GetLogger(), txDecoder, cosmosClient, block, blockResults, accountAddressPrefix, possibleSignerAddresses)
```
Comment (davcrypto) [inline usecase/parser/block.go:52]: pass a new logger instead

File: bootstrap/syncmanager.go [hunk 5]
```diff
@@ -221,6 +221,7 @@ func (manager *SyncManager) syncBlockWorker(blockHeight int64) ([]command_entity
 	}
 
 	commands, err := parser.ParseBlockToCommands(
+		manager.logger,
```
Comment (davcrypto) [inline bootstrap/syncmanager.go:224]: can pass new logger with block height field

File: usecase/parser/authinfo.go [hunk 6]
```diff
@@ -5,22 +5,31 @@ import (
...
+	txHash string,
 ) ([]model.TransactionSigner, error) {
 	var signers []model.TransactionSigner
 
 	if len(signerInfos) <= 0 && len(possibleSignerAddresses) <= 0 {
-		panic("error signer info not found")
+		return nil, fmt.Errorf("error signer info not found at tx %q", txHash)
+	}
+
+	if len(possibleSignerAddresses) == 1 {
```
Comment (davcrypto) [inline usecase/parser/authinfo.go:29]: shall we check if one of possibleSignerAddresses is empty or all of possibleSignerAddresses is empty?

File: usecase/parser/authinfo.go [hunk 7]
```diff
@@ -5,22 +5,31 @@ import (
...
 	signerInfos []utils.SignerInfo,
 	accountAddressPrefix string,
 	possibleSignerAddresses []string,
+	txHash string,
 ) ([]model.TransactionSigner, error) {
 	var signers []model.TransactionSigner
 
 	if len(signerInfos) <= 0 && len(possibleSignerAddresses) <= 0 {
-		panic("error signer info not found")
+		return nil, fmt.Errorf("error signer info not found at tx %q", txHash)
```
Comment (davcrypto) [inline usecase/parser/authinfo.go:26]: if returning error here, would un-parsed message type break this? if yes, suggest to log only

File: bootstrap/syncmanager.go (hunk 5 above)
Comment (vincentysc) [reply bootstrap/syncmanager.go:224]: or I should use logger directly?

File: usecase/parser/block.go [hunk 8]
```diff
@@ -48,8 +50,10 @@ func ParseBlockToCommands(
 		if parseErr != nil {
 			return nil, fmt.Errorf("error parsing message commands: %v", parseErr)
 		}
-
-		transactionCommands, parseErr := ParseTransactionCommands(txDecoder, cosmosClient, block, blockResults, accountAddressPrefix, possibleSignerAddresses)
+		logger = logger.WithFields(applogger.LogFields{
```
Comment (davcrypto) [inline usecase/parser/block.go:53]: create a new logger, you are assigning the logger in `ParseBlockToCommands`

**Review by davcrypto — APPROVED** [review]

**Pull request merged**